File: patterns/c-patterns/pattern42.c
// Pattern 42: ...

// 55555
// ****
// 333
// **
// 1

#include <stdio.h>
int main()
{
    int i, j;
    int n = 5;

    for (i = n; i >= 1; i--)
    {
        for (j = 1; j <= i; j++)
        {
            if (i % 2 == 0)
                printf("*");
            else
                printf("%d", i);
        }
        printf("\n");
    }
    return 0;
}

File: patterns/javascript-patterns/pattern42.js
// Pattern 42: ...

// 55555
// ****
// 333
// **
// 1

const n = 5;

for (let i = n; i >= 1; i--) {
    let row = '';

    for (let j = 1; j <= i; j++) {
        if (i % 2 === 0) {
            row += '*';
        } else {
            row += i;
        }
    }

    console.log(row);
}
